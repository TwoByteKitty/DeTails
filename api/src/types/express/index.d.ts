import { JwtPayload } from 'jsonwebtoken';

export {};

declare global {
  namespace Express {
    export interface Request {
      userToken?: string | JwtPayload;
    }
  }
}
