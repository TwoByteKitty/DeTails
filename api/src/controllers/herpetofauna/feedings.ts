import { Request, Response } from 'express';
import { Feeding, IMeal } from '../../models/herpetofauna/feeding';

const addFeeding = (request: Request<{ petId: string }, {}, IMeal>, response: Response) => {
  const newFeeding: IMeal = request.body;
  newFeeding.petId = request.params.petId;
  Feeding.create(newFeeding)
    .then((createdFeeding: IMeal) => response.json(createdFeeding))
    .catch((err: any) => response.status(500).json(err));
};

const getAllFeedingsByPet = (request: Request<{ petId: string }>, response: Response) => {
  Feeding.find({ petId: request.params.petId })
    .sort({ date: -1 })
    .then((foundFeedings: Array<IMeal>) => response.json(foundFeedings))
    .catch((err: any) => response.status(422).json(err));
};

const getSingleFeeding = (request: Request<{ feedingId: string }>, response: Response) => {
  Feeding.findById(request.params.feedingId)
    .then((foundFeeding: any) => response.json(foundFeeding))
    .catch((err: any) => response.status(422).json(err));
};

//needs more work!
// const editFeeding = (request: Request<{}, {}, IMeal>, response: Response) => {
//   const pet: IMeal = request.body;
//   pet.dateOfBirth = new Date(pet.dateOfBirth);
//   Feeding.findByIdAndUpdate(pet._id, pet)
//     .then((updatedFeeding) => response.json(updatedFeeding))
//     .catch((err) => response.status(500).json(err));
// };

export { getAllFeedingsByPet, addFeeding, getSingleFeeding };
