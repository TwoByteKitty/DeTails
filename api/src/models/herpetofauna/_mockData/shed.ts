import { IShed } from "../shed";

export const _shed: IShed = {
    _id: '507f191e810c19729de860ea',
    pinkBelly: "10/10/2022",
    blueEyes: "10/15/2022",
    clearEyes: "10/20/2022",
    shedSkin: "10/25/2022",
    entire: "Yes",
    shedComments: "test",
    petId: "507f191e810c19339de860ea"
};
