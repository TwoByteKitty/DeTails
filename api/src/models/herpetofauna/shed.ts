import { Schema, model, Date } from 'mongoose';

export interface IShed {
  _id: string;
  pinkBelly: string;
  blueEyes: string;
  clearEyes: string;
  shedSkin: string;
  entire: string;
  shedComments: string;
  petId: string;
}

export const shedSchema = new Schema<IShed>(
  {
    pinkBelly: { type: String, required: true },
    blueEyes: { type: String, required: true },
    clearEyes: { type: String, required: true },
    shedSkin: { type: String, required: true },
    entire: { type: String, required: true },
    shedComments: { type: String, required: false },
    petId: { type: String, required: true },
  },
  { toJSON: { virtuals: true } }
);

export const Shed = model<IShed>('Shed', shedSchema);
