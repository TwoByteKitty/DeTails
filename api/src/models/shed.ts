import { Schema, model, Date } from 'mongoose';

export interface IShed {
  _id: string;
  pinkBelly: Date;
  blueEyes: Date;
  clearEyes: Date;
  shedSkin: Date;
  entire: string;
  shedComments: string;
  petId: string;
}

export const shedSchema = new Schema<IShed>(
  {
    pinkBelly: { type: Date, required: true },
    blueEyes: { type: Date, required: true },
    clearEyes: { type: Date, required: true },
    shedSkin: { type: Date, required: true },
    entire: { type: String, required: true },
    shedComments: { type: String, required: false },
    petId: { type: String, required: true },
  },
  { toJSON: { virtuals: true } }
);

export const Shed = model<IShed>('Shed', shedSchema);
