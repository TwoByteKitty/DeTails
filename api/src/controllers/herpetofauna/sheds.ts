import { Request, Response } from 'express';
import { Shed, IShed } from '../../models/herpetofauna/shed';

const addShed = (request: Request<{ petId: string }, {}, IShed>, response: Response) => {
  const newShed: IShed = request.body;
  newShed.petId = request.params.petId;
  Shed.create(newShed)
    .then((createdShed: IShed) => response.json(createdShed))
    .catch((err: any) => response.status(500).json(err));
};

const getAllShedsByPet = (request: Request<{ petId: string }>, response: Response) => {
  Shed.find({ petId: request.params.petId })
    .sort({ date: -1 })
    .then((foundSheds: Array<IShed>) => response.json(foundSheds))
    .catch((err: any) => response.status(422).json(err));
};

const getSingleShed = (request: Request<{ shedId: string }>, response: Response) => {
  Shed.findById(request.params.shedId)
    .then((foundShed: any) => response.json(foundShed))
    .catch((err: any) => response.status(422).json(err));
};

//needs more work!
// const editShed = (request: Request<{}, {}, IShed>, response: Response) => {
//   const pet: IShed = request.body;
//   pet.dateOfBirth = new Date(pet.dateOfBirth);
//   Shed.findByIdAndUpdate(pet._id, pet)
//     .then((updatedShed) => response.json(updatedShed))
//     .catch((err) => response.status(500).json(err));
// };

export { getAllShedsByPet, addShed, getSingleShed };
