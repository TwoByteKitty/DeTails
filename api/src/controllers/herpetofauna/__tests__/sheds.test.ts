import { Shed } from "@api/models/herpetofauna/shed";
import { _shed } from "@api/models/herpetofauna/_mockData/shed";
import { getFindOneMock } from "@test/helpers/QueryMock";
import { Request, Response } from 'express';
import { createRequest, createResponse, MockRequest, MockResponse } from 'node-mocks-http';
import { getSingleShed } from "../sheds";

const mockingoose = require("mockingoose");

describe('Shed CRUD Controller', () => {
    let mockResponse: MockResponse<Response>;

    beforeEach(() => {
        mockResponse = createResponse();
    });

    it('getSingleShed: should return the doc with findById', async () => {
        const finderMock = getFindOneMock(_shed, '507f191e810c19729de860ea');
        const mockRequest: MockRequest<Request<{ shedId: string }>> = createRequest({ params: { shedId: '507f191e810c19729de860ea' } })

        mockingoose(Shed).toReturn(finderMock, 'findOne');
        await getSingleShed(mockRequest, mockResponse);

        expect(mockResponse._getJSONData()).toMatchObject(_shed);
    });

    it('getSingleShed: should return error status and and message', async () => {
        const finderMock = getFindOneMock(_shed, '507f191e810c19729de860ea');
        const mockRequest: MockRequest<Request<{ shedId: string }>> = createRequest({ params: { shedId: '507f191e810c19729de860ea' } })

        mockingoose(Shed).toReturn(new Error('Test Error'), 'findOne');
        await getSingleShed(mockRequest, mockResponse);

        expect(mockResponse.statusCode).toBe(422);
    });
});