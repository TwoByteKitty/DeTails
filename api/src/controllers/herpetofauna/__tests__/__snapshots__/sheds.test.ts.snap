// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shed CRUD Controller getSingleShed: should return the doc with findById: findById query 1`] = `
{
  "_id": "507f191e810c19729de860ea",
}
`;
